### DISCUSSION TITLE
Android循环模式下卡顿

### DISCUSSION THREAD

**Discussion by jsyjst** [Tencent/libpag#1696] [Q&A] [answered]
【版本信息】
PAG SDK 版本：4.3.3

【平台信息】
Android 原生
手机机型：Oppo Reno6 Pro 5G
Android版本：Android 13

【预期的表现】
用法比较简单，就是简单的播放一个pag文件，然后通过setRepeatCount(ValueAnimator.INFINITE)设置循环模式，能够比较流畅的播放动画

【实际的情况】
播放一次的时候是流畅的，但是播完重新播的时候会明显的卡顿，观看perfdog数据可以发现播放30s的卡顿率到达了16.9%，并且重新播的fps比较低。测试发现跟展示尺寸无关，跟机型相关，目前只有这台机型有问题，希望帮忙解答下～
<img width="1112" alt="image" src="https://github.com/Tencent/libpag/assets/41035185/12e40177-9b15-48e4-b988-57c730ffcabe">

【Demo及附件】
perfdog数据链接：https://perfdog.qq.com/case_detail/3937832
[卡顿pag文件](https://github.com/Tencent/libpag/files/12470792/2.pag.zip)
[卡顿视频.zip](https://github.com/Tencent/libpag/files/12470895/default.zip)

**Comment by domchen** [comment, marked as answer]
应该是因为视频序列帧的解码器初始化，你可以用PAGImageView来播放就不会卡顿了，或者开启一下PAGView.useDiskCache

**Reply by jsyjst** [reply]
感谢回复～，两种方式都试了。但是发现开启PAGView.useDiskCache是无效的还是卡顿，替换PAGImageView播放则有效。有3个问题想请教下哈，谢谢！
1. 如果是视频序列帧的解码器初始化导致的，是特定机型解码器初始化比较慢吗，因为只有部分机型才会卡顿。
2. PAGView内部是每次循环播放都会进行解码器初始化吗，不会复用吗？
3. 看PAGImageView的介绍，是做了磁盘缓存，目前我们PAG场景尺寸大概是150*150dp，整个文件2s，并且循环持续10s左右，会不会对性能有影响呢？
<img width="959" alt="image" src="https://github.com/Tencent/libpag/assets/41035185/c8bc4359-fbea-4425-840c-1cd14ef300fc">

**Reply by domchen** [reply]
1.视频是优先用硬件解码器，会和设备相关。
2.理论上是会复用的，循环的情况也会。具体要看你是不是用了组合增加了长度导致有一段时间不显示关闭了复用。要分析demo才能知道没有命中的原因。useDiskCache失败也是要看demo。
3.你这个面积极其小，720P以下面积都不用担心，是完全正向的收益，远比实时播放快很多。又流畅又省电。

**Reply by jsyjst** [reply]
好嘞谢谢。对于2还是有些疑问，这里demo用法很简单，就是简单的播放一个pag文件，代码没有用组合，具体使用如下面截图。播放的原始pag文件在问题附件里，这里会是pag原始文件有问题吗？
<img width="556" alt="image" src="https://github.com/Tencent/libpag/assets/41035185/f51b4207-53c3-4163-a4f8-b3eb7807b86f">
